browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -62,19 +105,96 @@ bool IPFSTabHelper::MaybeCreateForWebContents(
   return true;
 }
 
+void IPFSTabHelper::HostResolvedCallback(
+    const std::string& host,
+    const std::vector<std::string>& text_results) {
+  std::string dnslink = GetDSNRecordValue(text_results, kDnsLinkHeader);
+  if (dnslink.empty())
+    return;
+  GURL current = web_contents()->GetURL();
+  if (current.host() != host)
+    return;
+  ipfs_resolved_host_ = host;
+}
+
+GURL IPFSTabHelper::GetIPFSResolvedURL() const {
+  if (ipfs_resolved_host_.empty())
+    return GURL();
+  GURL current = web_contents()->GetURL();
+  GURL::Replacements replacements;
+  replacements.SetSchemeStr(kIPNSScheme);
+  return current.ReplaceComponents(replacements);
+}
+
+bool IPFSTabHelper::HasActiveResolverForHost(const std::string& host) const {
+  for (auto& resolver : resolvers_) {
+    if (!resolver->IsRunning())
+      continue;
+    if (resolver->host() == host)
+      return true;
+  }
+  return false;
+}
+
+void IPFSTabHelper::ResolveIPFSLink() {
```

Comment:
should this return early if the resolve method is IPFSResolveMethodTypes::IPFS_DISABLED ?